### PR TITLE
Remove mockRDSClient need for Endpoints map

### PR DESCRIPTION
**Pull request by ifosch**
Requires #124 to be merged

### CODE REVIEW

**Review by miguelbernadi — APPROVED** [review]
+1, but review my comments in case it can be simplified. Waiting for rebase to merge.

**Comment by ifosch** [comment]
@miguelbernadi I reviewed your comments for this PR and for #126:
* For this one's comment, something else instead of these would be better, I'll take some other approach in the incoming PR.
* For #126's comment about breaking the loop, I'll take it into the incoming PR.

I already rebased this, but I'll wait for your review before merging.

**Pull request merged**
